.github/workflows/build-and-test.yml

```diff
@@ -196,7 +196,7 @@ jobs:
       - name: Check long description with twine
         run: twine check $(find dist -name "natcap[._-]invest*")
 
-      - uses: actions/upload-artifact@v3
+      - uses: actions/upload-artifact@v4
         with:
           name: Source distribution

```

Comment:
It looks like there's a breaking change here: https://github.com/actions/upload-artifact?tab=readme-ov-file#breaking-changes

I guess we'll have to rename the artifact. Using `matrix.os` and `matrix.python-version` should do it.